presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
Will that work in the case where `grouping` uses `Identifier` and `group by` uses `DereferenceExpression` referring to the same column? Like in:
```
select ... grouping(a)
from ...
group by t.a
```

if not, #7354 will help.